pss/message/message.go

```diff
@@ -0,0 +1,46 @@
+package message
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+// Message encapsulates messages transported over pss.
+type Message struct {
+	To      []byte
+	Flags   Flags
+	Expire  uint32
+	Topic   Topic
+	Payload []byte
+}
+
+const digestLength = 32 // byte length of digest used for pss cache (currently same as swarm chunk hash)
+
+// Digest holds the digest of a message used for caching
+type Digest [digestLength]byte
+
+// New creates a new PSS message
+func New(flags Flags) *Message {
+	return &Message{
+		Flags: flags,
+	}
+}
+
+// Digest computes a message digest for use as a cache key
+func (msg *Message) Digest() Digest {
```

Comment:
Please consider adding expire to the digest. That will allow intentional resends of messages.

Reply:
Not part of this PR. This PR is a refactor and does not add functionality. `Expire` was not part of the digest.

Reply:
Added #1736 to track this. Should be written and prioritized as a user story.